### PR TITLE
Use darling and derive more things

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18208)

### CODE REVIEW

File: components/style_derive/cg.rs [hunk 1]
```diff
@@ -136,6 +137,16 @@ where
     }
 }
 
+pub fn parse_variant_attrs<A>(variant: &Variant) -> A
+where
+    A: FromVariant,
+{
+    match A::from_variant(variant) {
+        Ok(attrs) => attrs,
+        Err(e) => panic!("failed to parse attributes: {}", e),
```
Comment (emilio) [inline components/style_derive/cg.rs]: nit: Just `expect`?

File: components/style/properties/longhand/box.mako.rs [hunk 2]
```diff
@@ -370,11 +370,12 @@ ${helpers.single_keyword("position", "static absolute relative fixed",
...
         /// here.
         #[allow(non_camel_case_types)]
-        #[derive(Clone, Copy, Debug, PartialEq)]
         #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        #[derive(Animate, Clone, ComputeSquaredDistance, Copy, Debug, PartialEq)]
         pub enum T {
             % for keyword in vertical_align_keywords:
-                ${to_rust_ident(keyword)},
+            #[animation(error)]
+            ${to_rust_ident(keyword)},
```
Comment (emilio) [inline components/style/properties/longhand/box.mako.rs]: Is this intended? Like, I know it preserves the existing behaviour, but...

**Review by emilio — APPROVED** [review]

File: components/style_derive/cg.rs (hunk 1 above)
Comment (nox) [inline components/style_derive/cg.rs]: With `expect`:

```
    = help: message: failed to parse attributes: Error { kind: Multiple([Error { kind: MissingField("function"), locations: [] }, Error { kind: MissingField("comma"), locations: [] }]), locations: []
```

With an explicit panic and `Display`:

```
    = help: message: failed to parse attributes: Missing field `comma
```

The actual code if it was important for performances should be:

```rust
#[inline(never)]
#[cold]
fn panic(e: Error) -> ! {
    panic!("failed to parse attributes: {}", e)
}

match A::from_variant(variant) {
    Ok(attrs) => attrs,
    Err(e) => panic(e),
}
```

File: components/style/properties/longhand/box.mako.rs (hunk 2 above)
Comment (nox) [inline components/style/properties/longhand/box.mako.rs]: It is, the [spec](https://drafts.csswg.org/css-transitions/#animatable-css) says the property is animatable "as length".

Comment (emilio) [inline components/style/properties/longhand/box.mako.rs]: Maybe a comment would be nice?

Comment (nox) [inline components/style/properties/longhand/box.mako.rs]: Added one.

**Pull request merged**